AgniMessenger/src/test/AgniTestUtilities.java

```diff
@@ -0,0 +1,13 @@
+package test;
+
+public class AgniTestUtilities {
+    public static byte[] hexStringToByteArray(String s) {
+        int len = s.length();
+        byte[] data = new byte[len / 2];
+        for (int i = 0; i < len; i += 2) {
+            data[i / 2] = (byte) ((Character.digit(s.charAt(i), 16) << 4)
+                                 + Character.digit(s.charAt(i+1), 16));
+        }
```

Comment:
Apparently online I saw there was issues with leading zeroes, and I saw this function everywhere when I was googling how to do it so I felt like trusting the crowd on this one. Thanks for reading my PR :)